src/GraphQL.Tests/Types/SchemaTests.cs

```diff
@@ -125,10 +126,10 @@ public void handles_cycle_field_type()
     }
 
     [Fact]
-    public void handles_stackoverflow_exception_for_cycle_field_type()
+    public void handles_cycle_field_type_2()
     {
         var schema = new ACyclingDerivingSchema(new FuncServiceProvider(t => t == typeof(AbstractGraphType) ? new ConcreteGraphType() : null));
-        Should.Throw<InvalidOperationException>(() => schema.AllTypes["abcd"]);
+        schema.Initialize();
```

Comment:
It is unclear why there was cycle checks during initialization previously, or why this test failed.  The new code handles cycles within types properly and this schema can be initialized.